lib/chef/resource/aws_elasticsearch_domain.rb

```diff
@@ -0,0 +1,44 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+module AWS
+  class Elasticsearch
+    class Domain
+    end
+  end
+end
+
+class Chef::Resource::AwsElasticsearchDomain < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type ::AWS::Elasticsearch::Domain
+
+  INSTANCE_TYPES = %w(m3.medium.elasticsearch m3.large.elasticsearch
```

Comment:
If the user specifies an invalid instance_type, what does the error look like from AWS?  I prefer to let these kind of validations fall through to AWS (if their error is informative) so we don't have to keep our code in sync with their updated instance types.